### PR TITLE
feat(issue-priority): Rename Priority sort to Trends in UI

### PR DESCRIPTION
**Pull request by snigdhas**
Part 2/4 for https://github.com/getsentry/sentry/issues/65208:

Rename the Priority sort option to Trends to avoid confusing it with the new Group.priority column and feature. This PR changes the frontend option to use trends. The frontend will now send 'trends' for the sort type. The backend knows how to handle this and will convert 'priority' to trends to remain compatible with saved searches.

### CODE REVIEW

**Review by leeandher — APPROVED** [review]

**Review by jangjodi — APPROVED** [review]

**Review by malwilley — APPROVED** [review]
Looks like there's one test still referencing the old name.

We also might want to change the backend to accept `trends` to avoid confusion

**Comment by snigdhas** [comment]
Poking into it a bit more this might require a few steps to do safely
- update the BE to accept 'trends' and translate 'priority' -> 'trends'
- migrate rows with 'priority' in SavedSearch to 'trends'
- remove the 'priority' -> 'trends' translation

**Comment by malwilley** [comment]
@snigdhas that seems like a good approach! The only other thing would be to change the frontend to use `trends` in the URL and API request before taking away the translation (which may be as simple as changing this line https://github.com/getsentry/sentry/pull/65210/files#diff-21efbcbf0871789d984356191f4b570cd9f9cff6cf695c6b448394c205ef2667R171)

**Comment by snigdhas** [comment]
@malwilley rerequesting review as a sanity check since the backend has changed. Tested locally and existing saved searches with priority work as well as new searches with 'trends'.

**Review by malwilley — APPROVED** [review]

File: static/app/views/issueList/actions/sortOptions.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ function IssueListSortOptions({onSelect, sort, query}: Props) {
     ...(FOR_REVIEW_QUERIES.includes(query || '') ? [IssueSortOptions.INBOX] : []),
     IssueSortOptions.DATE,
     IssueSortOptions.NEW,
-    IssueSortOptions.PRIORITY,
```
[the remark this answers is not in the document]
Comment (snigdhas) [reply static/app/views/issueList/actions/sortOptions.tsx:40]: will wait to merge this until the migration in https://github.com/getsentry/sentry/pull/65311 is ready

**Comment by snigdhas** [comment]
The migration PR is merged and deployed, so I'll merge this and wait for it to deploy to all regions and then run the migration

**Pull request merged**